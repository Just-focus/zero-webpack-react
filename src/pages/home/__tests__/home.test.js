import { cleanup, fireEvent, render } from '@testing-library/react';
import CheckboxWithLabel from '../index';

afterEach(cleanup);

it('CheckboxWithLabel changes the text after click', () => {
	const { asFragment, queryByLabelText, getByLabelText } = render(
		<CheckboxWithLabel labelOn="On" labelOff="Off" />
	);

	expect(asFragment()).toMatchSnapshot();

	expect(queryByLabelText(/off/i)).toBeTruthy();

	fireEvent.click(getByLabelText(/off/i));

	expect(queryByLabelText(/on/i)).toBeTruthy();
});
