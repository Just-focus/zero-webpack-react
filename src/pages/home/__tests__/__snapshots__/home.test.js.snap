// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel changes the text after click 1`] = `
<DocumentFragment>
  <label>
    <input
      type="checkbox"
    />
    Off
  </label>
</DocumentFragment>
`;
